### DISCUSSION TITLE
target:add("configfiles", 绝对路径的情况下，无法生成指定名称的配置名

### DISCUSSION THREAD

**Discussion by ThzShun** [xmake-io/xmake#4133] [Q&A]
### Xmake 版本

2.8.1

### 操作系统版本和架构

windows10

### 描述问题

在自定义模块中
```
function setup(target)
    target:set("configvar", "License", "MIT")
    
    -- # C.1 绝对路径
    --local configFile = path.join("$(projectdir)", "/config.h.in")
    --local configFile = "$(projectdir)" .. "/config.h.in"

    -- # C.2 相对路径
    local configFile = "config.h.in"
    target:add("configfiles", configFile, {filename = "myconfig.h"})
end
```
只有相对路径的时候，生成才符合期望。

### 期待的结果

在build文件夹生成myconfig.h

### 工程配置

[复现demo](https://github.com/Wzshun/xmakeConfigDemo)

屏蔽C.2变量名，打开C.1的变量名，即可看到结果只生成了config.h，而不是myconfig.h

### 附加信息和错误日志

无

[a comment by an automated account is not shown]

**Comment by waruqi** [comment]
`$(projectdir)` 内建变量主要是给描述域用的，简化配置。。都在脚本域里面干了，就不用内建变量了，脚本域直接用 `path.join(os.projectdir(), "config.h.in")`

[a comment by an automated account is not shown]
